### PR TITLE
Django 4.2

### PR DESCRIPTION
**Pull request by katdom13**
This PR:
- Upgrades Django from ~3.2 to ~4.2
- Upgrades other packages
- Adds new migration files after Django upgrades

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -96,7 +96,7 @@ test_python:
   stage: test
   services:
     # Make sure this matches the Postgres version you run on your servers.
-    - postgres:9.6
+    - postgres:12.3
```
Comment (nickmoreton) [inline .gitlab-ci.yml:99]: **comment** Just noting that this change is OK but the file isn't used for CI

File: rca/settings/base.py [hunk 2]
```diff
@@ -785,3 +785,9 @@
     ENQUIRE_TO_STUDY_DESTINATION_EMAILS = env.get(
         "ENQUIRE_TO_STUDY_DESTINATION_EMAILS"
     ).split(",")
+
+
+# Silenced system checks
+SILENCED_SYSTEM_CHECKS = [
+    "templates.E003",  # Warns if two templatetags have the same name
+]
```
Comment (nickmoreton) [inline rca/settings/base.py:793]:
```suggestion
```

**suggestion** I think it's better to see the warning and at some point we should investigate a fix where this is seen.

File: .circleci/config.yml [hunk 3]
```diff
@@ -9,10 +9,11 @@ jobs:
           ALLOWED_HOSTS: localhost
           DATABASE_URL: postgresql://root@localhost/circle_test?sslmode=disable
           BIRDBATH_REQUIRED: false
-      - image: circleci/postgres:11.5
+      - image: circleci/postgres:12.3
```
Comment (nickmoreton) [inline .circleci/config.yml:12]:
```suggestion
      - image: circleci/postgres:12.17
```

**suggestion** The version on production is using 12.17 so maybe the docker-compose.yml file DB version could be updated too.

**Review by nickmoreton — CHANGES_REQUESTED** [review]
Hi @katdom13 I left a few suggestions.

**Comment by nickmoreton** [comment]
Hi @katdom13 This is now rebased on master.

**Comment by katdom13** [comment]
> I also saw this error when running collectstatic:
> 
> ```
> Traceback (most recent call last):
>   File "manage.py", line 10, in <module>
>     execute_from_command_line(sys.argv)
>   File "/venv/lib/python3.8/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
>     utility.execute()
>   File "/venv/lib/python3.8/site-packages/django/core/management/__init__.py", line 436, in execute
>     self.fetch_command(subcommand).run_from_argv(self.argv)
>   File "/venv/lib/python3.8/site-packages/django/core/management/base.py", line 412, in run_from_argv
>     self.execute(*args, **cmd_options)
>   File "/venv/lib/python3.8/site-packages/django/core/management/base.py", line 458, in execute
>     output = self.handle(*args, **options)
>   File "/venv/lib/python3.8/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 209, in handle
>     collected = self.collect()
>   File "/venv/lib/python3.8/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 154, in collect
>     raise processed
> whitenoise.storage.MissingFileError: The file 'js/sockjs.js.map' could not be found with <whitenoise.storage.CompressedManifestStaticFilesStorage object at 0xffff7c71e880>.
> 
> The JS file 'js/main.js' references a file which could not be found:
>   js/sockjs.js.map
> 
> Please check the URL references in this JS file, particularly any
> relative paths which might be pointing to the wrong location.
> ```
> 
> It doesn't show when running `fab build` but could do with an investigation I think. I have a feeling it will fail when deployed.

The `main.js` that is flagged seems to be a script produced by the webpack config. It might be tricky to check this one out.

**Comment by nickmoreton** [comment]
> > I also saw this error when running collectstatic:
> > ```
> > Traceback (most recent call last):
> >   File "manage.py", line 10, in <module>
> >     execute_from_command_line(sys.argv)
> >   File "/venv/lib/python3.8/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
> >     utility.execute()
> >   File "/venv/lib/python3.8/site-packages/django/core/management/__init__.py", line 436, in execute
> >     self.fetch_command(subcommand).run_from_argv(self.argv)
> >   File "/venv/lib/python3.8/site-packages/django/core/management/base.py", line 412, in run_from_argv
> >     self.execute(*args, **cmd_options)
> >   File "/venv/lib/python3.8/site-packages/django/core/management/base.py", line 458, in execute
> >     output = self.handle(*args, **options)
> >   File "/venv/lib/python3.8/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 209, in handle
> >     collected = self.collect()
> >   File "/venv/lib/python3.8/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py", line 154, in collect
> >     raise processed
> > whitenoise.storage.MissingFileError: The file 'js/sockjs.js.map' could not be found with <whitenoise.storage.CompressedManifestStaticFilesStorage object at 0xffff7c71e880>.
> > 
> > The JS file 'js/main.js' references a file which could not be found:
> >   js/sockjs.js.map
> > 
> > Please check the URL references in this JS file, particularly any
> > relative paths which might be pointing to the wrong location.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > It doesn't show when running `fab build` but could do with an investigation I think. I have a feeling it will fail when deployed.
> 
> The `main.js` that is flagged seems to be a script produced by the webpack config. It might be tricky to check this one out.

Lets see if it deploys OK before we spend more time on it. Unless you think it could delay things later on?

**Pull request merged**